docs/interfacing_with_surrogates.rst

```diff
@@ -0,0 +1,250 @@
+    Interfacing with non-differentiable, non-JITtable models is possible (for an example, see the `QuaLiKiz transport model`_) if the linear solver is used. However, note that if the model is called within the step function JIT will need to be disabled with ``TORAX_COMPILATION_ENABLED=0``.
+
+
+Option 1: manually reimplementing the model in JAX
+==================================================
+
+If the architecture of the surrogate is sufficiently simple, you might consider reimplementing the model in JAX.
+The surrogates in TORAX are mostly implemented using `Flax Linen`_, and can be found in the |fusion_surrogates|_ repository.
+If you're not familiar with Flax, you can check out the `Flax documentation`_ on how to define your own models.
+
+Consider a PyTorch neural network,
+
+.. code-block:: python
+
+    import torch
+
+    class PyTorchMLP(torch.nn.Module):
+      def __init__(self, hidden_dim: int, n_hidden: int, output_dim: int, input_dim: int):
+        super().__init__()
+        self.model = torch.nn.Sequential(
+          torch.nn.Linear(input_dim, hidden_dim),
+          torch.nn.ReLU(),
+          *[torch.nn.Sequential(
+            torch.nn.Linear(hidden_dim, hidden_dim),
+            torch.nn.ReLU()
+          ) for _ in range(n_hidden)],
+          torch.nn.Linear(hidden_dim, output_dim)
+        )
+
+      def forward(self, x):
+        return self.model(x)
+
+    torch_model = PyTorchMLP(hidden_dim, n_hidden, output_dim, input_dim)
+
+This model can be converted to a Flax model as follows:
+
+.. code-block:: python
+
+    from flax import linen
+
+    class FlaxMLP(linen.Module):
+      hidden_dim: int
+      n_hidden: int
+      output_dim: int
+      input_dim: int
+
+    @linen.compact
+    def __call__(self, x):
+      x = linen.Dense(self.hidden_dim)(x)
+      x = linen.relu(x)
+      for _ in range(self.n_hidden):
+        x = linen.Dense(self.hidden_dim)(x)
+        x = linen.relu(x)
+      x = linen.Dense(self.output_dim)(x)
+      return x
+
+    flax_model = FlaxMLP(hidden_dim, n_hidden, output_dim, input_dim)
+
+As this is only the model architecture, we need to load the trained weights separately.
+This can be a bit fiddly as you have to map from the parameter names in the weights checkpoint file to the parameter names in the Flax model.
+
+For loading weights from a PyTorch checkpoint, you might do something like:
+
+.. code-block:: python
+
+    import torch
+
+    state_dict = torch.load(PYTORCH_CHECKPOINT_PATH)
+
+    params = {}
+    for i in range(n_hidden_layers):
+      layer_dict = {
+        "kernel": jnp.array(
+          state_dict[f"model.{i*2}.weight"]
+        ).T,
+        "bias": jnp.array(
+          pytorch_state_dict[f"model.{j*2}.bias"]
+        ).T,
+      }
+      params[f"Dense_{i}"] = layer_dict
+
+    params = {'params': params}
+
+
+The model can then be called like any Flax model,
```

Comment:
Have you verified this? Eg. that no params are transposed between libraries

Reply:
Yes, verified on our TGLFNN model.
I've added a warning here to encourage the user to be aware that intricacies like this may exist